src/core/utils/checks/charge_neutrality.hpp

```diff
@@ -1,33 +1,33 @@
 #include <cmath>
-
-#include "PartCfg.hpp"
+#include <limits>
 
 namespace Utils {
 
-inline void check_charge_neutrality(PartCfg &partCfg) {
-  double total_charge = 0.0;
-  double min_charge = 0.0;
-  for (auto const &particle : partCfg) {
-    total_charge += particle.p.q;
-    double abs_charge = std::abs(particle.p.q);
-    if (min_charge == 0.0 and abs_charge != 0.0)
-      min_charge = abs_charge;
-    if (abs_charge < min_charge and abs_charge != 0.0)
-      min_charge = particle.p.q;
+template<typename ParticleRange, typename Particle>
+inline void check_charge_neutrality(ParticleRange &prange) {
+  std::vector<double> charges;
+  std::transform(prange.begin(), prange.end(), std::back_inserter(charges),
+          [](Particle const &p){return p.p.q;});
+  double total_charge = std::accumulate(charges.begin(), charges.end(), 0.0);
+  std::sort(charges.begin(), charges.end(),
+          [](double a, double b) {return std::abs(a) < std::abs(b);});
+  //Find first non-zero element.
+  auto result = std::find_if(charges.begin(), charges.end(), [](double a){return std::abs(a) > std::numeric_limits<double>::epsilon();});
```

Comment:
```cpp
return std::abs(a) > 0.0;
```
You're looking at values here, not at differences.  What if the smallest charge is `1e-30`?

Reply:
then you have other issues anyways